transfer/multichain_queue_test.go

```diff
@@ -0,0 +1,102 @@
+package transfer
+
+import (
+	"errors"
+	"math/big"
+	"testing"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/core/types"
+	"github.com/stretchr/testify/assert"
+)
+
```

Comment:
We should have two tests:
* Make multiple queues and test that the correct transactions was sent to the correct queue. (something like you got there with the `green path`)
* Send a transaction for a queue that doenst exist - make sure it returns an error.

Tests like `t.Run("exec produced an error", func(t *testing.T)`  or  `t.Run("closed queue returns an error", func(t *testing.T)` actually test the underlying queue and not the multiqueue and thats not needed.

Reply:
@tomasmik I did this but I'm not sure if it's correct